src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -93,9 +98,27 @@ private Func<
         private volatile int _disposed;
         private SafeWinHttpHandle? _sessionHandle;
         private readonly WinHttpAuthHelper _authHelper = new WinHttpAuthHelper();
+        private readonly Timer? _certificateCleanupTimer;
+        private bool _isTimerRunning;
+        private readonly ConcurrentDictionary<CachedCertificateKey, CachedCertificateValue> _cachedCertificates = new();
 
         public WinHttpHandler()
         {
+            if (CertificateCachingAppContextSwitchEnabled)
+            {
+                WeakReference<WinHttpHandler> thisRef = new(this);
+                _certificateCleanupTimer = new Timer(
```

Comment:
We'll want to suppress the ExecutionContext during the timer allocation to avoid capturing unrelated asynclocals.

Since this is targeting framework as well, it'd be something like this:
```c#
bool restoreFlow = false;
try
{
    if (!ExecutionContext.IsFlowSuppressed())
    {
        ExecutionContext.SuppressFlow();
        restoreFlow = true;
    }
    
    _ = new Timer(...);
}
finally
{
    if (restoreFlow)
    {
        ExecutionContext.RestoreFlow();
    }
}
```